### DISCUSSION TITLE
Looking for MVVM example on WinUI3 Desktop.

### DISCUSSION THREAD

**Discussion by vmx17** [CommunityToolkit/dotnet#71] [General]
Hi! Are there any example project that contains WindowsAppSDK(1.0-) + multi-windowing + MVVM + UnitTtest + paging (add/remove/unify/separate pages in NavigationView on multiple windows) support gracefully? I think these might be the template for new desktop applications in Microsoft's way. Though I tried to make some prototypes but miserable for myself. I'd like to move to WinUI3 from WindowsForms but short of XAML knowledge.

**Comment by LeftTwixWand** [comment]
Hey, I'm working on a new open source project, that will be similar to [Microsoft/InventorySample](https://github.com/microsoft/inventorysample), that built on UWP (WinUI 2).
The new one will be a really modern one, using WinUI 3, and it's the latest features.
But I will need a few more weeks to bring this to a minimum working version.

**Reply by vmx17** [reply]
How's going? I've managed to made a sample using MVVM elements but still too far from my purpose.
And now I'm in head of DirectX12 on WinUI3 with C++/WinRT and should place it as C#-based XAML SwapChain. Though Microsoft has released some articles about interop DirectX11 with XAML in C++/CX, still DirectX team releasing example on Win32. Please re-union!

**Reply by LeftTwixWand** [reply]
Hey, now I'm trying to set up unit tests, integration tests and UI test (using win app driver).
It's a bit difficult, so I've spent a lot of time to get into. Because there are so normal projects, can show how to do it correct.

My version of the Inventory, you can check here: [github.com/LeftTwixWand/Inventory](https://github.com/LeftTwixWand/Inventory)
Also, I've made some changes in the app architecture.
Because the official one is using Service architecture, that is really common, and it's easy to find a good sample.
But I want to show how the real production app have built. 
So I've added kind of DDD + SQRS, and now the app has much more layers:
**Application, Infrastructure, Domain, Presentation, Persistence**
